configure.ac

```diff
@@ -1,189 +0,0 @@
-# removed when this behavior becomes the default.
-# tar-ustar is required because there may be files whose entire paths exceed
-# 99 characters.
-AM_INIT_AUTOMAKE([-Wall -Wno-portability foreign 1.11 parallel-tests
-    subdir-objects tar-ustar])
-# Avoid spewing garbage over the terminal ('make V=1' to see the garbage)
-AM_SILENT_RULES([yes])
-# Keep Autotools macros local to this source tree
-AC_CONFIG_MACRO_DIR([m4])
-# Use Libtool
-LT_INIT
-
-AX_IS_RELEASE([git-directory])
-
-AC_CACHE_SAVE
-
-# Variables to define
-# -------------------
-
-# Required versions of libraries
-# Update these whenever you use a function that requires a certain API version.
-# Keep these lists sorted alphabetically.
-EOS_METRICS_REQUIREMENT="eosmetrics-0 >= 0.2"
-GIO_REQUIREMENT="gio-unix-2.0 >= 2.46"
-GLIB_REQUIREMENT="glib-2.0 >= 2.63.1"
-GOBJECT_REQUIREMENT="gobject-2.0 >= 2.58"
-OSTREE_REQUIREMENT="ostree-1 >= 2013.7"
-POLKIT_REQUIREMENT="polkit-gobject-1"
-SOUP_REQUIREMENT="libsoup-2.4 >= 2.42"
-SQLITE3_REQUIREMENT="sqlite3"
-UUID_REQUIREMENT="uuid"
-EMER_REQUIRED_MODULES="$EOS_METRICS_REQUIREMENT \
-                       $GIO_REQUIREMENT \
-                       $GLIB_REQUIREMENT \
-                       $GOBJECT_REQUIREMENT \
-                       $OSTREE_REQUIREMENT \
-                       $POLKIT_REQUIREMENT \
-                       $SOUP_REQUIREMENT \
-                       $SQLITE3_REQUIREMENT \
-                       $UUID_REQUIREMENT"
-EMER_SHARED_REQUIRED_MODULES="$GIO_REQUIREMENT \
-                              $GOBJECT_REQUIREMENT \
-                              $SQLITE3_REQUIREMENT \
-                              $UUID_REQUIREMENT"
-AC_SUBST(EMER_REQUIRED_MODULES)
-AC_SUBST(EMER_SHARED_REQUIRED_MODULES)
-
-AC_DEFINE([GLIB_VERSION_MIN_REQUIRED], [GLIB_VERSION_2_64], [Require at least this GLib version])
-AC_DEFINE([GLIB_VERSION_MAX_ALLOWED], [GLIB_VERSION_2_64], [Prevent using GLib API newer than this version])
-
-# Required build tools
-# --------------------
-# Make sure we can create directory hierarchies
-AC_PROG_MKDIR_P
-# C compiler
-AC_PROG_CC
-# Make sure the C compiler supports per-target CFLAGS
-AC_PROG_CC_C_O
-# Make sure the C compiler supports C99
-AC_PROG_CC_C99
-# Library configuration tool
-PKG_PROG_PKG_CONFIG
-# DBus code generation tool
-AC_PATH_PROG([GDBUS_CODEGEN], [gdbus-codegen], [notfound])
-AS_IF([test "x$GDBUS_CODEGEN" = "xnotfound"],
-    [AC_MSG_ERROR([Could not find gdbus-codegen])])
-AC_ARG_VAR([GDBUS_CODEGEN], [Path to gdbus-codegen])
-
-AC_CACHE_SAVE
-
-
-# Configure options
-# -----------------
-# --enable-compile-warnings. By default, warnings are always fatal. Documentation:
-# https://www.gnu.org/software/autoconf-archive/ax_compiler_flags.html
-#
-# -Wno-declaration-after-statement:
-#   Mixed decls are used a lot and are harmless.
-AX_COMPILER_FLAGS([WARN_CFLAGS], [WARN_LDFLAGS], [no], [],
-                  [-Wno-declaration-after-statement])
```

Comment:
I guess dropping this option is covered by enabling `c_std=c11` in the Meson project options?

You may want to consider also setting `warning_level=3` (or some other value) in Meson’s `project()`.

Reply:
Yes, I think so. Certainly I don't get warned about mixed declarations.

- `warning_level=3` enables `-Wpedantic` which is upset about some indeed pedantic stuff which I would rather not fix today.
- `warning_level=2` enables `-Wextra`  which spews warnings about unused parameters; adding `-Wno-unused-parameter` makes the build clean. So I'll set this option.

It doesn't seem straightforward to preserve the "fatal warnings for git checkout, non-fatal otherwise" default.